File: Selenium/ElementalSelenium/UploadFileTest.py
from selenium import webdriver
import unittest
import os
import HTMLReport
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait


class UploadFileTest(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        cls.driver = webdriver.Firefox()
        cls.driver.set_window_position(2000, 0)
        cls.driver.maximize_window()

    @classmethod
    def tearDownClass(cls):
        cls.driver.close()

    @classmethod
    def test_00_upload_file(self):
        self.driver.get("http://the-internet.herokuapp.com/upload")
        file = open("../README.md")
        path = os.path.realpath("../README.md")
        filename = os.path.basename("../README.md")
        file.close()
        self.driver.find_element_by_id("file-upload").send_keys(path)
        self.driver.find_element_by_id("file-submit").click()

        WebDriverWait(self.driver, 10).until(
            expected_conditions.presence_of_element_located((By.ID, "uploaded-files")))
        result = self.driver.find_element_by_id("uploaded-files").text
        print(result)
        print(filename)
        # self.assertTrue(expected_conditions.presence_of_element_located(By.ID, ""))
        assert result == filename
        # self.assertEqual(result, filename)
        # self.assertEqual(result, filename)


if __name__ == '__main__':
    unittest.main(testRunner=HTMLReport.TestRunner(
        report_file_name="Report_Upload",
        output_path="../../Reports",
        lang="en"
    ))

File: Selenium/ElementalSelenium/UploadFileTest.java
import org.openqa.selenium.By;
import org.openqa.selenium.Point;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.*;

import java.io.File;

@Test
public class UploadFileTest {
    private static WebDriver driver;

    @BeforeClass
    public void setUp() throws Exception {
        System.setProperty("webdriver.gecko.driver","Drivers/geckodriver.exe");
        driver = new FirefoxDriver();
        driver.manage().window().setPosition(new Point(2000,0));
        driver.manage().window().maximize();
        driver.get("http://the-internet.herokuapp.com/upload");
    }

    @Test
    public void uploadElementExists() throws Exception{
        driver.findElement(By.id("file-upload"));
    }

    @Test
    public void submitElementExists() throws Exception{
        driver.findElement(By.xpath("//*[@id=\"file-submit\"]\n"));
    }

    @Test
    public void uploadFile() throws Exception{
        File file = new File("Selenium/README.md");
        String path = file.getAbsolutePath();
        driver.findElement(By.id("file-upload")).sendKeys(path);
        driver.findElement(By.xpath("//*[@id=\"file-submit\"]\n")).click();
        String actual = driver.findElement(By.id("uploaded-files")).getText();
        Assert.assertEquals(actual,file.getName());
    }

    @AfterClass
    public void tearDown() throws Exception{
        driver.quit();
    }


}
